### ISSUE TITLE
Run ROS demos

### ISSUE THREAD

**Issue by gawa-K** [gawa-K/2018summer_vacation_project#2]
@k-okada
@Affonso-Gui

**Comment by Affonso-Gui** [comment]
For now:
- http://wiki.ros.org/ROS/Tutorials/UnderstandingTopics
- http://wiki.ros.org/ROS/Tutorials/UnderstandingServicesParams

Then maybe:
- http://wiki.ros.org/ROS/Tutorials/CreatingPackage
- http://wiki.ros.org/ROS/Tutorials/WritingPublisherSubscriber%28c%2B%2B%29 (C++)
- http://wiki.ros.org/ROS/Tutorials/WritingPublisherSubscriber%28python%29 (python)

**Comment by Affonso-Gui** [comment]
@gawa-K Any progress?

**Comment by gawa-K** [comment]
@Affonso-Gui I have no progress. I will have time to make progress in a few days.

**Comment by Affonso-Gui** [comment]
Please reopen this issue and leave it open until you finish the demos (after a few days or even after a few weeks, no need to hurry).

**Comment by gawa-K** [comment]
I can't execute "Using rostopic echo (2.3)"

**Comment by gawa-K** [comment]
@Affonso-Gui I can't execute "Using rostopic echo (2.3)"

**Comment by gawa-K** [comment]
The error is "you may only specify one input topic"

**Comment by gawa-K** [comment]
@Affonso-Gui The error is "you may only specify one input topic"

**Comment by Affonso-Gui** [comment]
You are probably giving more than one argument to `rostopic echo`.
Open three terminals and run the following commands (one command in each terminal):
```
roscore
```
```
rosrun turtlesim turtlesim_node
```
```
rostopic echo /turtle1/pose 
```
The final command should tell you the pose of the turtle.


Also, please leave this issue opened. (Click on `Reopen` now, and when commenting use `Comment` and not `Close and Comment`).

**Comment by gawa-K** [comment]
Certainly, "rostopic echo /turtle1/pose" told me the pose of the turtle.
However, "rostopic echo / turtle1 / cmd_vel" says the error "WARNING: topic [/turtle1/cmd_vel] does not appear to be published yet" and "rostopic echo / turtle1 / command_velocity" says the error "WARNING: topic [/ turtle1 / command_velocity] does not appear to be published yet"

**Comment by Affonso-Gui** [comment]
> "rostopic echo / turtle1 / cmd_vel" says the error "WARNING: topic [/turtle1/cmd_vel] does not appear to be published yet"

This is not an error, it is only a warning telling you that there are no new messages on this topic. Try `rostopic info /turtle1/cmd_vel`, it will show you that there are indeed no publishers registered to this topic.

Later on the tutorial we have this command:
```
rostopic pub -1 /turtle1/cmd_vel geometry_msgs/Twist -- '[2.0, 0.0, 0.0]' '[0.0, 0.0, 1.8]'
```
Try to execute the above while running `rostopic echo /turtle1/cmd_vel` on another terminal. It will show you the published message.

Also, **be aware of spaces**. Topic names should not have any spaces before or after the slashes. Using wrong spaces will lead to the `you may only specify one input topic` error.

> "rostopic echo / turtle1 / command_velocity"

This topic is not registered. You can see the list of registered topics with `rostopic list`.

**Comment by gawa-K** [comment]
"rostopic pub -1 /turtle1/cmd_vel geometry_msgs/Twist -- '[2.0, 0.0, 0.0]' '[0.0, 0.0, 1.8]'" was successed but "rostopic pub -1 /turtle1/cmd_vel geometry_msgs/Twist -r 1 - '[2.0, 0.0, 0.0]' '[0.0, 0.0, 1.8]'" said the error"You cannot select both -r and -1".
Also, rqt_graph express only two circle(/turtlesim and /rostopic_7921_1536309262273) and an arrow(/turtle1/pose).

**Comment by Affonso-Gui** [comment]
Error messages are not your enemy. Read them carefully and they will tell you how to solve your problem.

Here it clearly says: 
```
You cannot select both -r and -1
```

Indeed, you are using both `-r` and `-1` in your command.  Remove either `-1` or `-r 1` from your statement and it will work again.

Many times you can get more information on commands by adding `--help` to it. 
Executing `rostopic pub --help` will tell you that `-1` means _publish once_, while `-r 1` means _publish every second_. Since these two options are contradictory, setting both of them rightfully leads to an error.
 
```
  -r RATE, --rate=RATE  publishing rate (hz).  For -f and stdin input, this
                        defaults to 10.  Otherwise it is not set.
  -1, --once            publish one message and exit
```

You don't need to care much about `rqt_graph` for now.

**Comment by gawa-K** [comment]
I finished the session2(http://wiki.ros.org/ROS/Tutorials/UnderstandingServicesParams)

**Comment by Affonso-Gui** [comment]
Great.
Go ahead to try some real coding in the next tutorials.

- http://wiki.ros.org/ROS/Tutorials/CreatingPackage
- http://wiki.ros.org/ROS/Tutorials/WritingPublisherSubscriber%28c%2B%2B%29 (C++)
  OR
- http://wiki.ros.org/ROS/Tutorials/WritingPublisherSubscriber%28python%29 (python)

**Comment by gawa-K** [comment]
The command "mkdir -p /catkin_ws/src" said "mkdir: cannot create directory '/carkin_ws': Permission denied".
I check existing the directory "/catkin_ws" by using the command "ls", but the directory doesn't exist.

**Comment by Affonso-Gui** [comment]
Try removing the slash bar:
```
mkdir -p catkin_ws/src
```

`/` is the root directory, so you need admin permission to change files in it. The following will also work (although it is not recommended)
```
sudo mkdir -p /catkin_ws/src
```
